completions/fio

```diff
@@ -70,12 +77,70 @@ _fio()
             _filedir -d
             return
             ;;
+        --ioengine)
+            _comp_cmd_fio__engines "$1"
+            COMPREPLY=($(compgen -W '${ret[@]}' -- "$cur"))
+            return
+            ;;
+        --exec_postrun | --exec_prerun)
+            COMPREPLY=($(compgen -c -- "$cur"))
+            return
+            ;;
+        --uid)
+            _uids
+            return
+            ;;
+        --gid)
+            _gids
+            return
+            ;;
+        --?*)
+            local IFS=$'\n'
+            local cmdhelp=($("$1" --cmdhelp=${prev#--} 2>/dev/null))
+            _comp_unlocal IFS
+            case ${cmdhelp[*]} in
+                *"showing closest match"*)
+                    # ignore
+                    ;;
+                *" type: boolean "* | *" type: empty or boolean "*)
+                    COMPREPLY=($(compgen -W '0 1' -- "$cur"))
+                    return
+                    ;;
+                *" valid values:"*)
+                    # For example, for --kb_base=:
+                    #        valid values: 1024       [...]
+                    #                    : 1000       [...]
+                    local line="" in_values=false
+                    for line in "${cmdhelp[@]}"; do
+                        if $in_values; then
+                            if [[ $line =~ ^[[:space:]]*:[[:space:]]*([^[:space:]]+) ]]; then
+                                ret+=("${BASH_REMATCH[1]}")
+                            else
+                                break
+                            fi
+                        elif [[ $line =~ ^[[:space:]]*valid\ values:[[:space:]]*([^[:space:]]+) ]]; then
+                            in_values=true
+                            ret+=("${BASH_REMATCH[1]}")
+                        fi
+                    done
+                    COMPREPLY=($(compgen -W '${ret[@]}' -- "$cur"))
```

Comment:
```suggestion
                    COMPREPLY=($(compgen -W '"${ret[@]}"' -- "$cur"))
```

It is not necessary in the present case because the elements of `ret` do not contain spaces, but that is what we would know only after we carefully read the code.